src/app/Http/Controllers/PostController.php

```diff
@@ -62,19 +61,33 @@ public function viewPost(Post $post): View
     /**
      * Like Post
      */
-    public function like(Post $post): RedirectResponse
+    public function like(Post $post): JsonResponse
     {
         $this->postService->like($post);
-        return redirect()->back();
+        $likeUsers = $post->likes->take(self::USER_LIKED)->map(function ($like) {
+            return $like->user->username;
+        });
+    
+        return response()->json([
+            'likes_count' => $post->likes->count(),
+            'like_users' => $likeUsers,
+        ]);
     }
-
+    
     /**
-     * Unlike Post
+     * UnLike Post
      */
-    public function unlike(Post $post): RedirectResponse
+    public function unlike(Post $post): JsonResponse
```

Comment:
I dont think this returns JsonReponse alone, line 84 you are returning `$like->user->username;`

Reply:
joke tama pala

Reply:
The return $like->user->username; is inside the map function which then transforms the value of $post->likes. This then transferred into a variable that is returned using json response